library/objective-c/EnvoyEngineImpl.m

```diff
@@ -29,13 +117,33 @@ - (void)dealloc {
   [[NSNotificationCenter defaultCenter] removeObserver:self];
 }
 
+- (int)registerFilter:(EnvoyHTTPFilter *)filter {
+  // TODO(goaway): Everything here leaks, but it's all be tied to the life of the engine.
+  // This will need to be updated for https://github.com/lyft/envoy-mobile/issues/332
```

Comment:
Assuming my comment above about providing an instance of each filter for each request, I think this will become more problematic, no?

Reply:
Yes. :)

It will need to be solved if/when we go that route.